### PR TITLE
Sidearms for Combine soldiers who were tricked into being unarmed

### PR DESCRIPTION
**Pull request by Blixibon**
**This is for after the Mapbase v7.0 merge!!!**

---

At the moment, players can use the soldier weapon-swapping feature to trick squadmates into being unarmed. This feature allows those squadmates to pull out a pistol if they enter combat.

### CODE REVIEW

File: sp/src/game/server/hl2/npc_combine.cpp [hunk 1]
```diff
@@ -1076,6 +1076,13 @@ void CNPC_Combine::Weapon_Drop( CBaseCombatWeapon *pWeapon, const Vector *pvecTa
 		// Make sure they're allowed to re-arm themselves in case they fail to pick up this weapon.
 		m_bTemporarilyNeedWeapon = true;
 		m_bDontPickupWeapons = false;
+
+		// Check if we don't have other weapons in our inventory
+		if ( !m_hMyWeapons[1].Get() )
+		{
+			// In case we get into combat while unarmed, have a pistol on-hand
+			GiveWeaponHolstered( AllocPooledString( "weapon_pistol" ) );
```
Comment (1upD) [inline sp/src/game/server/hl2/npc_combine.cpp:1085]: This is a bit goofy, but could we make it a pulse pistol for Bad Cop (and maaaaaybe regular Combine elites?) and a 357 for the nemesis (if for some unforeseen reason he is an ally)

**Review by 1upD — CHANGES_REQUESTED** [review]
I hate to mark this as 'needs changes' over something so trivial but I had one suggestion I'd like to get your comment on before we go ahead with this change.

File: sp/src/game/server/hl2/npc_combine.cpp (hunk 1 above)
Comment (Blixibon) [reply sp/src/game/server/hl2/npc_combine.cpp:1085]: In my opinion, that is something which should be done through the `GiveWeaponHolstered` input on a map-by-map or `ent_fire`-by-`ent_fire` basis. You can give Bad Cop, elites, the nemesis, etc. any weapon you want for their inventories and they will pull them out automatically in the same way as this method.

Also, I don't think E:Z2's 357 worldmodel works with NPCs at the moment.

Comment (1upD) [reply sp/src/game/server/hl2/npc_combine.cpp:1085]: I don't know. We're not going to be going through maps for this. It seems to me like elites should get pulse pistols if regular soldiers get USPs.

Does that sound okay? If you seriously object, I'll go ahead and give approval

**Review by 1upD — APPROVED** [review]
I'm going to approve this. I had commented and requested pulse pistols for elites but that can be handled separately.

**Pull request merged**